keymutex/hashed.go

```diff
@@ -42,17 +42,17 @@ type hashedKeyMutex struct {
 
 // Acquires a lock associated with the specified ID.
 func (km *hashedKeyMutex) LockKey(id string) {
-	km.mutexes[km.hash(id)%len(km.mutexes)].Lock()
+	km.mutexes[km.hash(id)%uint(len(km.mutexes))].Lock()
 }
 
 // Releases the lock associated with the specified ID.
 func (km *hashedKeyMutex) UnlockKey(id string) error {
-	km.mutexes[km.hash(id)%len(km.mutexes)].Unlock()
+	km.mutexes[km.hash(id)%uint(len(km.mutexes))].Unlock()
 	return nil
 }
 
-func (km *hashedKeyMutex) hash(id string) int {
+func (km *hashedKeyMutex) hash(id string) uint {
```

Comment:
it's fine to cast from uint32 to uint, but I'm not sure why we're doing that.

Reply:
hi @apelisse 
The [issue](https://github.com/kubernetes/kubernetes/issues/73858) mentions that on arm machine, the output of `int(h.Sum32())` maybe negative, and cause problems in `LockKey` and `UnlockKey` due to this negative number will be used as the slice index.

But I'm not sure, whether convert to `uint` is safe here.

Otherwise, `math.Abs` could be another option for this case ?

Reply:
Yeah, sorry I made myself unclear. I meant, why don't you just return an uncasted uint32?

Reply:
aha, that's right. I miss the return type of `h.Sum32()`. Thanks.
Already updated, PTAL.